platform/android/app/src/main/kotlin/coop/polypoly/polypod/network/Network.kt

```diff
@@ -85,8 +86,19 @@ open class Network(val context: Context) {
         authToken: String?,
     ): HttpURLConnection? {
         var connection: HttpURLConnection
+        var requestURL: URL
         try {
-            connection = URL(url).openConnection() as HttpURLConnection
+            requestURL = URL(url)
+        } catch (e: MalformedURLException) {
+            logger.error(e.toString())
+            return null
+        }
+        if (requestURL.protocol != "https") {
+            logger.error("network.$type failed, URL is not secure (https)")
+            return null
```

Comment:
What about returning an error to the feature, like you do on iOS?

Reply:
Returning null here will return an error to the feature using the "throw a generic error to the feature when the response is null" mechanism

Reply:
the null returned here will throw an error in endpoint, in iOs, although we do return an error, that error is only used for logging and gets replaced by a generic error in postoffice. Functionally they pretty much do the same thing

Reply:
Got it!